honeybee_radiance/cli/multiphase.py

```diff
@@ -505,3 +507,179 @@ def _aperture_view_factor(project_folder, apertures, size=0.2, ambient_division=
+    'to yield a final number of grids to generate. This is useful in workflows where '
+    'the grid-count is being interpreted as a cpu-count but there are multiple '
+    'processors acting on a single grid. To ignore this limitation set the value '
+    'to 1. Default: 1.', type=int, default=1)
+@click.option(
+    '--min-sensor-count', '-msc', help='Minimum number of sensors in each output grid. '
+    'Use this number to ensure the number of sensors in output grids never gets very '
+    'small. This input will override the input grid-count when specified. '
+    'To ignore this limitation, set the value to 1. Default: 1.', type=int,
+    default=1)
+@click.option(
+    '--sun-path',
+    type=click.Path(exists=True, file_okay=True, dir_okay=False, resolve_path=True),
+    default=None, show_default=True,
+    help='Path for a sun-path file that will be added to octrees for direct sunlight '
+    'studies. If sunpath is provided an extra octree for direct_sun will be created.'
+)
+@click.option(
+    '--phase', type=click.Choice(['2', '3', '5']), default='5', show_default=True,
+    help='Select a multiphase study for which octrees will be created. 3-phase includes '
+    '2-phase, and 5-phase includes 3-phase and 2-phase.'
+)
+@click.option(
+    '--octree-folder', help='Output folder into which the octree files be written.',
+    default='octree', show_default=True)
+@click.option(
+    '--grid-folder', help='Output folder into which the grid files be written.',
+    default='grid', show_default=True)
+def octrees_grids(folder, grid_count, grid_divisor, min_sensor_count, sun_path, 
+                          phase, octree_folder, grid_folder):
+    """Generate a set of octrees and sensor grids from a folder.
+
+    This command will generate octrees for both default and direct studies. It will do so
+    for static apertures and aperture groups, creating one octree for each light path,
+    i.e., all other light paths are blacked.
+
+    Sensor grids will be redistributed if they are to be used in a two phase simulation.
+    A subfolder for each light path will be created. In this folder the redistributed
+    grids are found.
+
+    \b
+    Args:
+        folder: Path to a Radiance model folder.
+        grid_count: Number of output sensor grids to be created. This number
+            is usually equivalent to the number of processes that will be used to run
+            the simulations in parallel.
+    """
+    model_folder = ModelFolder.from_model_folder(folder)
+
+    # check if sunpath file exist - otherwise continue without it
+    if sun_path and not os.path.isfile(sun_path):
+        sun_path = None
+
+    if phase == '5' and not sun_path:
+        raise RuntimeError(
+            'To generated octrees for a 5 Phase study you must provide a sunpath.'
+        )
+
+    phases = {
+        '2': ['two_phase'],
+        '3': ['two_phase', 'three_phase'],
+        '5': ['two_phase', 'three_phase', 'five_phase']
+    }
+
+    try:
```

Comment:
This is becoming a really long CLI command.

What we usually do is to create separate smaller functions for each step so we can test them separately and then we use them in the CLI command. Can you think of a logical way to break this down and write a few tests for each part?